assets/model_monitoring/components/src/action_analyzer/action_analyzer_identify_problem_traffic/run.py

```diff
@@ -5,265 +5,124 @@
     TRACE_ID_COLUMN,
-    SPAN_ID_COLUMN,
-    ROOT_QUESTION_COLUMN,
-    TOPIC_LIST_COLUMN,
-    GROUP_LIST_COLUMN,
     VIOLATED_METRICS_COLUMN,
-    INDEX_CONTENT_COLUMN,
-    INDEX_SCORE_COLUMN,
-    INDEX_ID_COLUMN,
     ROOT_SPAN_COLUMN,
     GROUP_TOPIC_MIN_SAMPLE_SIZE,
-    RETRIEVAL_QUERY_TYPE_COLUMN,
-    RETRIEVAL_TOP_K_COLUMN,
-    GOOD_METRICS_VALUE,
-    DEFAULT_TOPIC_NAME
+    GOOD_METRIC_THRESHOLD,
+    BAD_METRIC_THRESHOLD,
+    DEFAULT_TOPIC_NAME,
+    GROUP_COLUMN,
+    QUERY_INTENTION_COLUMN,
+    GOOD_GROUP_NAME,
+    BAD_GROUP_NAME
 )
-from shared_utilities.prompts import BERTOPIC_DEFAULT_PROMPT
-from shared_utilities.span_tree_utils import SpanTree
+from action_analyzer.prompts import BERTOPIC_DEFAULT_PROMPT
 from model_data_collector_preprocessor.store_url import StoreUrl
 
 from shared_utilities.io_utils import (
     try_read_mltable_in_spark,
     save_spark_df_as_mltable,
-    save_empty_dataframe
+    create_spark_df
 )
 from shared_utilities.llm_utils import (
     API_KEY,
     _WorkspaceConnectionTokenManager
 )
-
-from bertopic import BERTopic
-from openai import AzureOpenAI
-from bertopic.representation import OpenAI
+try:
+    from bertopic import BERTopic
+    from openai import AzureOpenAI
+    from bertopic.representation import OpenAI
+except Exception:
+    print("Error while importing Bertopic")
 
 
 def get_output_schema() -> StructType:
     """Get Output Data Spark DataFrame Schema."""
     schema = StructType(
         [
             StructField(TRACE_ID_COLUMN, StringType(), True),
-            StructField(SPAN_ID_COLUMN, StringType(), True),
-            StructField(ROOT_QUESTION_COLUMN, StringType(), True),
-            StructField(PROMPT_COLUMN, StringType(), True),
-            StructField(COMPLETION_COLUMN, StringType(), True),
-            StructField(TOPIC_LIST_COLUMN, StringType(), True),
-            StructField(GROUP_LIST_COLUMN, StringType(), True),
-            StructField(VIOLATED_METRICS_COLUMN, StringType(), True),
-            StructField(INDEX_CONTENT_COLUMN, StringType(), True),
-            StructField(INDEX_ID_COLUMN, StringType(), True),
-            StructField(CONTEXT_COLUMN, StringType(), True),
-            StructField(INDEX_SCORE_COLUMN, FloatType(), True),
-            StructField(RETRIEVAL_QUERY_TYPE_COLUMN, StringType(), True),
-            StructField(RETRIEVAL_TOP_K_COLUMN, IntegerType(), True)
+            StructField(GROUP_COLUMN, StringType(), True),
+            StructField(QUERY_INTENTION_COLUMN, StringType(), True),
         ]
     )
     return schema
 
 
+def save_violated_metrics(violated_metrics, output_path):
+    """Save violated metrics into spark dataframe."""
+    schema = StructType(
+        [
+            StructField(VIOLATED_METRICS_COLUMN, StringType(), True)
+        ]
+    )
+    data = []
+    for metric in violated_metrics:
+        data.append([metric])
+    df = create_spark_df(data, schema)
+    save_spark_df_as_mltable(df, output_path)
```

Comment:
in long term, we'd like to deprecate mltable dependency. use it only if you have to.